state/application_test.go

```diff
@@ -1696,10 +1715,56 @@ func (s *ApplicationSuite) TestForceDoesNotAllowRemovingAnApplicationWithOffers(
 	op := s.mysql.DestroyOperation()
 	op.Force = true
 	err := s.State.ApplyOperation(op)
-	c.Assert(err, gc.ErrorMatches, `cannot destroy application "mysql": application is used by 1 offer`)
+	c.Assert(err, jc.ErrorIsNil)
 	assertOffersRef(c, s.State, "mysql", 1)
```

Comment:
this can't be right since removing the app should remove the offers too (and the code appears to suggest it does but something is wrong if this test passes)

Reply:
> Unless I'm missing something the key driver we discussed which was the reason to adopt txn rev no checking is not being tested

`TestRemovingApplicationFailsRace` should address this